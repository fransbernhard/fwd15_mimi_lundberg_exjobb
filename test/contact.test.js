import React, { Component } from 'react';
import Contact from '../src/components/contact';
import renderer from 'react-test-renderer';
// import {shallow} from 'enzyme';

describe('components', () => {
  describe('<Contact/>', () => {
    it('renders correctly', () => {
      const tree = renderer.create(<Contact />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});

test('Input field updates on msg input', () => {
  const component = renderer.create(
    <Contact />
  );
  component
    .getInstance()
    ._handleChangeMsg({target: { value: "New msg"}});
  // re-rendering
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})


test('Input field updates on email input', () => {
  const component = renderer.create(
    <Contact />
  );
  component
    .getInstance()
    ._handleChange({target: { value: "New email"}});
  // re-rendering
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

//
// test('Input field updates on input', () => {
//   const component = shallow(
//      <Contact />
//   );
//
//   const input = component.find('#formEmail');
//   expect(input.text()).toEqual("");
//   // expect(input.text()).toEqual('');
//   input.simulate('change',
//     {target: { value: "My new value"}}
//   );
//   const val = input.node.value;
//
//   expect(input.text()).toEqual(val);
// });
