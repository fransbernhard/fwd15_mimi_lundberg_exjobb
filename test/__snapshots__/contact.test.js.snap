// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input field updates on email input 1`] = `
<div
  className="contact"
  id="contact"
>
  <div
    className="filter"
  >
    <form
      className="form"
      id="formContact"
      onSubmit={[Function]}
    >
      <label>
        Email
      </label>
      <input
        id="formEmail"
        name="formEmail"
        onChange={[Function]}
        required={true}
        type="email"
        value="New email"
      />
      <label>
        Meddelande
      </label>
      <textarea
        cols="40"
        id="formMsg"
        name="formMsg"
        onChange={[Function]}
        required={true}
        rows="8"
        value=""
      />
      <input
        className="btn--cta"
        id="btn-submit"
        type="submit"
        value="Submit"
      />
    </form>
  </div>
</div>
`;

exports[`Input field updates on msg input 1`] = `
<div
  className="contact"
  id="contact"
>
  <div
    className="filter"
  >
    <form
      className="form"
      id="formContact"
      onSubmit={[Function]}
    >
      <label>
        Email
      </label>
      <input
        id="formEmail"
        name="formEmail"
        onChange={[Function]}
        required={true}
        type="email"
        value=""
      />
      <label>
        Meddelande
      </label>
      <textarea
        cols="40"
        id="formMsg"
        name="formMsg"
        onChange={[Function]}
        required={true}
        rows="8"
        value="New msg"
      />
      <input
        className="btn--cta"
        id="btn-submit"
        type="submit"
        value="Submit"
      />
    </form>
  </div>
</div>
`;

exports[`components <Contact/> renders correctly 1`] = `
<div
  className="contact"
  id="contact"
>
  <div
    className="filter"
  >
    <form
      className="form"
      id="formContact"
      onSubmit={[Function]}
    >
      <label>
        Email
      </label>
      <input
        id="formEmail"
        name="formEmail"
        onChange={[Function]}
        required={true}
        type="email"
        value=""
      />
      <label>
        Meddelande
      </label>
      <textarea
        cols="40"
        id="formMsg"
        name="formMsg"
        onChange={[Function]}
        required={true}
        rows="8"
        value=""
      />
      <input
        className="btn--cta"
        id="btn-submit"
        type="submit"
        value="Submit"
      />
    </form>
  </div>
</div>
`;
